### ISSUE TITLE
old client users hitting usage errors when trying to use old-style cli commands with new ua client

### ISSUE THREAD

**Issue by blackboxsw** [canonical/ubuntu-advantage-client#909]
In the old ua-tools client, esm was enabled with:
`ubuntu-advantage enable-esm <esm_apt_user>:<esm_apt_passord>`

when running legacy subcommands on new client the system is not configured. This could break existing deployment scripts for customers using cloud-config or automation tooling:

```
root@dev-t:~# ua enable-esm asdf:asdf
usage: ua [command] [flags]
ua: error: argument : invalid choice: 'enable-esm' (choose from 'status', 'attach', 'attach-premium', 'detach', 'enable', 'disable', 'refresh', 'version', 'help')
```

This feature of compatible CLI was rejected in initial spec work for trusty. Does new ua client need to provide more helpful error messages when old subcommands are used:

Example:
```
# ua enable-esm asdf:asdf
The command provided is deprecated.
For updated attach instructions: see https://ubuntu.com/advantage
```

**Comment by mearns** [comment]
Is `enable` the replacement for `enable-esm`?

**Comment by powersj** [comment]
Hi @mearns 

> Is `enable` the replacement for `enable-esm`?

Yes, the new UA client that is now in Ubuntu 14.04 LTS (Trusty) and the upcoming Ubuntu 20.04 LTS (Focal) now uses the `sudo ua enable <service>` command to enable and disable services. This is done after attaching with the token via `sudo ua attach <token>`. No more needing to manage multiple PPA credentials for each service.

**Comment by jbidinger-datica** [comment]
What version of the ubuntu-advantage-tools package did the command options change? I need to update some automation and it would be nice to know what version I need to look for.

**Comment by powersj** [comment]
@jbidinger-datica any version > 18 has the new changes.

Currently the new version available in the archive for Ubuntu 14.04 LTS (Trusty) and the soon to be released Ubuntu 20.04 LTS (Focal). Updates to Ubuntu 16.04 LTS (Xenial) and 18.04 LTS (Bionic) are coming at a later date.

**Comment by blackboxsw** [comment]
I vote we shouldn't fix this as original spec for ua tools was that new client wouldn't support old client params. The issue we have is that the new-client requires a separate auth token mechanism to properly setup and maintain these subscriptions and the old-style credentials don't apply to this solution approach. See https://ubuntu.com/advantage for new token details.  Please re-open if this is a blocker.
